pkgs/data/fonts/nerdfonts-patcher/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, python3
+, fontforge
+
+, patchFont ? null
+, fontName ? null
+, glyphFlags ? "-c"
+}:
+
+assert (patchFont != null);
+assert (fontName != null);
+
+stdenv.mkDerivation rec {
+  pname = fontName + "-nerd-font";
+  version = "v2.1.0";
+  
+  src = fetchFromGitHub {
+    owner = "ryanoasis";
+    repo = "nerd-fonts";
+    rev = version;
+    sha256 = "sha256-j0cLZHOLyoItdvLOQVQVurY3ARtndv0FZTynaYJPR9E=";
+  };
+
+  nativeBuildInputs = [
+    python3
+    fontforge
+  ];
+
+  buildPhase = ''
```

Comment:
I would be grate to package also the program font-patcher itself.

Reply:
That's a great idea, since the name of the package would suggest that, although ad-hoc patching would go against nixos declarative approach.
I could maybe focus on packaging the patcher, (and name the output store path accordingly), and then somehow integrate font patching into it?

Reply:
Yes, 1. package the patcher, 2. provide a generator or wrapper derivation which can patch arbitrary font packages.